### ISSUE TITLE
yarn cache clean removes global packages

### ISSUE THREAD

**Issue by joscha** [yarnpkg/yarn#1137]
**Do you want to request a *feature* or report a *bug*?**

Bug, I think (could be intended behaviour, but should be documented in that case)

**What is the current behavior?**

`yarn cache clean` removes also dependencies that have been installed with `yarn global add` before.

**If the current behavior is a bug, please provide the steps to reproduce.**

```console
yarn global add lerna
lerna --version # prints version
yarn cache clean
lerna --version # fails
```

**What is the expected behavior?**

Cache should be emptied, but globally installed packages should persist.

**Please mention your node.js, yarn and operating system version.**
node 6.5.0
yarn 0.15.1
OSX 10.12 and alpine 3.4

**Comment by WYSIATI** [comment]
@joscha is your command `lerna --version`?

I tried to reproduced your problem but failed. My commands are:
`yarn global add lerna`
`lerna --version # -bash: lerna: command not found`
`yarn cache clean`
`lerna --version # -bash: lerna: command not found`

BTW, `-g, --global                DEPRECATED` is from `yarn --help`, the `--global` flag is deprecated now.

**Comment by joscha** [comment]
@WYSIATI I just tried it again:

```console
➜  yarn global add chalk-cli
yarn global v0.15.1
warning No license field
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Installed chalk-cli@3.0.0 with binaries:
      - chalk
✨  Done in 1.57s.
➜  chalk -t '{red Hello}'
Hello
➜  yarn cache clean
yarn cache v0.15.1
success Cleared cache.
✨  Done in 5.41s.
➜  chalk -t '{red Hello}'
zsh: command not found: chalk
```

**Comment by WYSIATI** [comment]
@joscha 

hmm, you are running with `zsh`, while on my mac with `bash`, I won't find the command even just added the package globally......

May I ask if you are running these on native docker for mac?

**Comment by joscha** [comment]
I noticed the problem in an Alpine 3.4 docker container

**Comment by victornoel** [comment]
@joscha are you sure this wasn't caused by the fact that with 0.15.1, the global folder is stored in the cache folder, which isn't the case anymore with 0.16.x! (i.e., you should test again with latest yarn :)

**Comment by joscha** [comment]
@victornoel:

```
➜  /tmp yarn --version
0.16.1
➜  /tmp yarn global add chalk-cli
yarn global v0.16.1
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 📃  Building fresh packages...
success Installed chalk-cli@3.0.0 with binaries:
      - chalk
✨  Done in 2.00s.
➜  /tmp chalk blue Test
Test
➜  /tmp yarn cache clean
yarn cache v0.16.1
success Cleared cache.
✨  Done in 1.54s.
➜  /tmp chalk blue Test
Test
```

seems to be fixed in `0.16.1`.
